core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -133,34 +226,98 @@ public GeoBoundingBoxQueryBuilder topRight(String geohash) {
      * Sets the type of executing of the geo bounding box. Can be either `memory` or `indexed`. Defaults
      * to `memory`.
      */
-    public GeoBoundingBoxQueryBuilder type(String type) {
-        this.type = type;
+    public GeoBoundingBoxQueryBuilder type(Type type) {
+        this.type = (type != null) ? type : Type.MEMORY;
         return this;
     }
 
+    /** Returns the execution type of the geo bounding box.*/
+    public Type type() {
+        return type;
+    }
+
+    /** Returns the name of the field to base the bounding box computation on. */
+    public String fieldName() {
+        return this.fieldName;
+    }
+
     @Override
-    protected void doXContent(XContentBuilder builder, Params params) throws IOException {
-        // check values
-        if(Double.isNaN(box[TOP])) {
-            throw new IllegalArgumentException("geo_bounding_box requires top latitude to be set");
-        } else if(Double.isNaN(box[BOTTOM])) {
-            throw new IllegalArgumentException("geo_bounding_box requires bottom latitude to be set");
-        } else if(Double.isNaN(box[RIGHT])) {
-            throw new IllegalArgumentException("geo_bounding_box requires right longitude to be set");
-        } else if(Double.isNaN(box[LEFT])) {
-            throw new IllegalArgumentException("geo_bounding_box requires left longitude to be set");
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (GeoUtils.isValidLatitude(topLeft.getLat()) == false) {
```

Comment:
For ```geo_point``` we currently support a ```normalize``` option, which, when set to ```false``` enables the user to index points outside the standard -180:180, -90:90 spatial coordinate system.  If ```normalize``` is set to false this will still validate and throw an unexpected exception? Also see updated PR #12300 which merges ```normalize_lat``` and ```normalize_lon``` into one option and adds a ```validate``` option to the query to let users explicitly specify whether they want validation.